src/libraries/System.Linq/src/System/Linq/Join.cs

```diff
@@ -7,9 +7,212 @@ namespace System.Linq
+        ///     List{Pet} pets = new List{Pet} { barley, boots, whiskers, daisy };
+        ///
+        ///     // Create a list of Person-Pet pairs where
+        ///     // each element is an anonymous type that contains a
+        ///     // Pet's name and the name of the Person that owns the Pet.
+        ///     var query =
+        ///         people.Join(pets,
+        ///             person => person,
+        ///             pet => pet.Owner,
+        ///             (person, pet) =>
+        ///                 new { OwnerName = person.Name, Pet = pet.Name });
+        ///
+        ///     foreach (var obj in query)
+        ///     {
+        ///         Console.WriteLine(
+        ///             "{0} - {1}",
+        ///             obj.OwnerName,
+        ///             obj.Pet);
+        ///     }
+        /// }
+        ///
+        /// /*
+        ///  This code produces the following output:
+        ///
+        ///  Hedlund, Magnus - Daisy
+        ///  Adams, Terry - Barley
+        ///  Adams, Terry - Boots
+        ///  Weiss, Charlotte - Whiskers
+        /// */
+        /// </code>
+        /// </example>
+        /// <remarks>
+        /// <para>
+        /// This method is implemented by using deferred execution. The immediate return value is an object that stores
+        /// all the information that is required to perform the action. The query represented by this method is not
+        /// executed until the object is enumerated either by calling its <c>GetEnumerator</c> method directly or by
+        /// using <c>foreach</c> in C# or <c>For Each</c> in Visual Basic.
+        /// </para>
+        /// <para>
+        /// The default equality comparer, <see cref="EqualityComparer{T}.Default" />, is used to hash and compare keys.
+        /// </para>
+        /// <para>
+        /// A join refers to the operation of correlating the elements of two sources of information based on a common key.
+        /// <see cref="Join{TOuter, TInner, TKey, TResult}(IEnumerable{TOuter}, IEnumerable{TInner}, Func{TOuter, TKey}, Func{TInner, TKey}, Func{TOuter, TInner?, TResult})" />
+        /// brings the two information sources and the keys by which they are matched together in one method call.
+        /// </para>
+        /// <para>
+        /// In relational database terms, the <see cref="Join{TOuter, TInner, TKey, TResult}(IEnumerable{TOuter}, IEnumerable{TInner}, Func{TOuter, TKey}, Func{TInner, TKey}, Func{TOuter, TInner, TResult})" /> method implements an inner equijoin.
+        /// 'Inner' means that only elements that have a match in the other sequence are included in the results.
+        /// An 'equijoin' is a join in which the keys are compared for equality.
+        /// An left outer join can be performed using the
+        /// <see cref="LeftJoin{TOuter, TInner, TKey, TResult}(IEnumerable{TOuter}, IEnumerable{TInner}, Func{TOuter, TKey}, Func{TInner, TKey}, Func{TOuter, TInner?, TResult})" /> method,
```

Comment:
I'm guessing it's failing because of the nullable annotations?

Reply:
Ah, did not know these weren't supported in XML docs. Will check and remove.